src/Tinker.jl

```diff
@@ -1,4 +1,268 @@
 module Tinker
 
-# package code goes here
+using Gtk.ShortNames, GtkReactive, Graphics, Colors, Images
+
+## Generally useful structs and functions
+# Rectangle structure
+struct Rectangle
+    x::Float64
+    y::Float64
+    w::Float64
+    h::Float64
+end
+
+Rectangle() = Rectangle(0,0,-1,-1)
+Base.isempty(R::Rectangle) = R.w < 0 || R.h < 0
+
+# rectangle draw function
+function drawrect(ctx, rect, color)
+    set_source(ctx, color)
+    rectangle(ctx, rect.x, rect.y, rect.w, rect.h)
+    stroke(ctx)
+end;
+
+# Zooms zr to the decimal % entered; view centered around center XY
+function zoom_percent(z::Float64, zr::ZoomRegion, center::XY{Int64})
```

Comment:
> does this do anything different from https://github.com/JuliaGizmos/GtkReactive.jl/blob/7f122fec26afe0efaeefd4f4024e39b2c34ac20e/src/graphics_interaction.jl#L416-L426

Yes! My zoom_percent function zooms based on the full view, meaning that no matter how far you're zoomed in, it will zoom _to_ the percent of the full view that the user gives it. GtkReactive's zoom zooms based on the current view, meaning that giving it a value of 0.5 at any zoom level will zoom in based on the current view. For my purposes, I need a zoom based on the full view.